python/triton/testing.py

```diff
@@ -110,7 +112,7 @@ def do_bench(fn, warmup=25, rep=100, grad_to_none=None, quantiles=None, return_m
     :param return_mode: The statistical measure to return. Options are "min", "max", "mean", "median", or "all" Default is "mean".    :type return_mode: str
     """
     assert return_mode in ["min", "max", "mean", "median", "all"]
-    import torch
+    import numpy as np
 
     di = runtime.driver.active.get_device_interface()
```

Comment:
It doesn't really remove the torch dependency.

`get_device_interface` still imports troch

Reply:
Depends on the driver. For cpu driver, it doesn't need torch, since there's currently only [one device](https://github.com/triton-lang/triton-cpu/blob/daa7eb01e1f3fe60d0e0b9a643886f32d3c3ffe7/third_party/cpu/backend/driver.py#L435-L436) (the host cpu).

Reply:
hmm. I thought we are still going to import torch for the CPU backend anyway.

@minjang can you confirm that you want the CPU runtime to be completely independent of torch?

Reply:
@Jokeren No, triton-cpu doesn't have a plan or need to reduce/remove torch dependency.

@stephen-huan, Okay, I understand your intention. It's okay to reduce the dependency only for `third_party/cpu`. But, as you already had to change `python/triton/testing.py`, you will need to change many parts of the code outside `third_party/cpu`. And, due to (painful) rebasing and resolving merge conflicts, triton-cpu strongly wants to avoid such code changes. For example, `test_core.py` has heavy mixed usages of torch and numpy. Even if this is for testing, we still anyhow have both torch and numpy dependencies. So, right now, I agree with @Jokeren.

Reply:
> But, as you already had to change `python/triton/testing.py`, you will need to change many parts of the code outside `third_party/cpu`. And, due to (painful) rebasing and resolving merge conflicts, triton-cpu strongly wants to avoid such code changes. For example, `test_core.py` has heavy mixed usages of torch and numpy. Even if this is for testing, we still anyhow have both torch and numpy dependencies.

Maybe there is a miscommunication somewhere? The goal of this PR is not to remove torch entirely from triton-cpu. My goal is for the code of an end user using triton to write kernels (e.g. myself) not to need to depend on torch at runtime when importing triton / excuting kernels / autotuning. The development dependencies of triton itself are irrelevant, since they aren't applicable to the dependencies of a triton end user.

> But, as you already had to change `python/triton/testing.py`, you will need to change many parts of the code outside `third_party/cpu`.

I don't think this is the case? The modified function `_summarize_statistics` (note the leading underscore) is a completely internal api that as far as I know is used exactly twice, both times in `python/triton/testing.py` itself. Indeed, the version of the PR that was in triton-cpu _did_ pass [all the tests](https://github.com/triton-lang/triton-cpu/actions/runs/12483579668/job/34842507197) without modifications (modulo the device parameter I accidentally left in). The current failure in https://github.com/triton-lang/triton-cpu/pull/205 is an unrelated failure in the CI.

Reply:
To give an analogy, triton depends on {sicpy,pandas,matplotlib} for the tests and certain benchmarking features (e.g. `perf_report`, printing out a table of times, plotting a graph of runtimes).

https://github.com/triton-lang/triton/blob/f8b5301a92459199e1b9faf7aadf1a7c10bb9866/README.md?plain=1#L133-L135

I would imagine it would be very difficult to remove these dependencies entirely and maintain the same functionality! However, if a user installing triton also _had_ to install scipy, pandas, matplotlib just to import triton and run kernels, that would be very strange and unexpected, especially if they weren't declared upfront.

Currently, triton depends on torch at runtime for a number of tasks, even importing triton will fail without torch. But this is _not explicitly declared_ (i.e. `pip install triton` will not install torch) because torch _itself_ cyclically depends on triton for `torch.compile`. This is perhaps unavoidable for the gpu backends without a lot of additional work, but for the cpu backend it _is_ easily avoided, and this is precisely the aim of this PR.

Reply:
> To give an analogy, triton depends on {sicpy,pandas,matplotlib} for the tests and certain benchmarking features (e.g. `perf_report`, printing out a table of times, plotting a graph of runtimes).
> 
> https://github.com/triton-lang/triton/blob/f8b5301a92459199e1b9faf7aadf1a7c10bb9866/README.md?plain=1#L133-L135
> 
> I would imagine it would be very difficult to remove these dependencies entirely and maintain the same functionality! However, if a user installing triton also _had_ to install scipy, pandas, matplotlib just to import triton and run kernels, that would be very strange and unexpected, especially if they weren't declared upfront.
> 
> Currently, triton depends on torch at runtime for a number of tasks, even importing triton will fail without torch. But this is _not explicitly declared_ (i.e. `pip install triton` will not install torch) because torch _itself_ cyclically depends on triton for `torch.compile`. This is perhaps unavoidable for the gpu backends without a lot of additional work, but for the cpu backend it _is_ easily avoided, and this is precisely the aim of this PR.

the file you are changing is for testing only, we do explicitly depend on torch for testing and the dependency still exist with your patch as mentioned by Keren so I don't understand how it helps. Also I don't think it should affect non GPU backends outside if the testing infrastructure?

Reply:
Is the [autotuning feature](https://triton-lang.org/main/python-api/generated/triton.autotune.html) considered "testing only"? I assumed `triton.autotune` should be part of the public api.

https://github.com/triton-lang/triton/blob/89ea017e2edeb5a424935bf1b81ffed0f3e3694c/python/triton/runtime/autotuner.py#L108-L123